### PR TITLE
Add deny access functionality

### PR DESCRIPTION
**Pull request by AnderssonPeter**
* Added logic to stop JsonPatch from using accessing specific properties
* Added documentation for DenyPatch
* Added editorconfig file and ensure that we use tabs instead of spaces
* Added unit tests

Resolves ##33

### CODE REVIEW

File: SystemTextJsonPatch/Internal/PropertyProxyCache.cs [hunk 1]
```diff
@@ -49,11 +51,20 @@ internal static class PropertyProxyCache
...
 					return new PropertyProxy(propertyInfo);
 				}
 			}
 
 			return null;
 		}
+
+		private static void EnsureAccessToProperty(PropertyInfo propertyInfo)
+		{
+			if (propertyInfo.GetCustomAttribute(typeof(DenyPatchAttribute), true) != null)
```
Comment (Havunen) [inline SystemTextJsonPatch/Internal/PropertyProxyCache.cs:64]: Wasn't there also generic overload `propertyInfo.GetCustomAttribute<DenyPatchAttribute>(...)` `?

**Comment by Havunen** [comment]
The changes looks good to me. Thank you!

**Review by Havunen — APPROVED** [review]

File: SystemTextJsonPatch/Internal/PropertyProxyCache.cs (hunk 1 above)
Comment (Havunen) [reply SystemTextJsonPatch/Internal/PropertyProxyCache.cs:64]: Its not a big deal anyway

Comment (AnderssonPeter) [reply SystemTextJsonPatch/Internal/PropertyProxyCache.cs:64]: Yes there is, but i don't think it exists in `netstandard2.0`.

**Pull request merged**